api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -660,19 +660,12 @@ public void changePassword(User user, String oldPassword, String newPassword) th
 		updatePassword(user, newPassword);
 	}
 	
-	/**
-	 * This is for internal use only. DO NOT CALL THIS METHOD.
-	 * 
-	 * @param user The user's password to change
-	 * @param newPassword The password to change it to
-	 */
-	@Authorized(PrivilegeConstants.EDIT_USER_PASSWORDS)
+	@Override
 	public void changePassword(User user, String newPassword) {
-		if (!Daemon.isDaemonThread() || !Context.getUserContext().getAuthenticatedUser().isSuperUser()) {
-			throw new APIAuthenticationException(Context.getMessageSourceService().getMessage("error.privilegesRequired",
-				new Object[] { "System Developer" }, Context.getLocale()));
+		if ("admin".equals(user.getUsername()) && Boolean.valueOf(Context.getRuntimeProperties()
```

Comment:
This check was removed from the updatePassword method, so I added it here to make the changePassword(User,String) method functionality equivalent to what is used to be.

Reply:
(I ended up pulling this out because this turned out to be functionality that had only been added as part of this ticket)